silver/models/payment_methods.py

```diff
@@ -126,14 +137,29 @@ def cancel(self):
 
         return None
 
-    def save(self, **kwargs):
-        self.clean()
-        super(PaymentMethod, self).save(**kwargs)
+    def clean_with_old_state(self, old_state):
```

Comment:
`previous_instance` ? In this project context, `state` can be easily confused with the field, not the object's state.